$OpenBSD: patch-Rakefile,v 1.1 2015/05/27 21:18:48 ajacoutot Exp $

build system: install desktop files and icons if Qt is enabled

--- Rakefile.orig	Sun May 10 05:04:26 2015
+++ Rakefile	Wed May 27 03:12:35 2015
@@ -56,7 +56,7 @@ def setup_globals
   $application_subdirs     =  { "mmg" => "mmg/", "mkvtoolnix-gui" => "mkvtoolnix-gui/" }
   $applications            =  $programs.collect { |name| "src/#{$application_subdirs[name]}#{name}" + c(:EXEEXT) }
   $manpages                =  $programs.collect { |name| "doc/man/#{name}.1" }
-  $manpages                << "doc/man/mkvtoolnix-gui.1" if !$build_mkvtoolnix_gui
+  $manpages                << "doc/man/mkvtoolnix-gui.1" if $build_mkvtoolnix_gui
 
   $system_includes         = "-I. -Ilib -Ilib/avilib-0.6.10 -Isrc"
   $system_libdirs          = "-Llib/avilib-0.6.10 -Llib/librmff -Lsrc/common"
@@ -563,7 +563,8 @@ end
 # Installation tasks
 desc "Install all applications and support files"
 targets  = [ "install:programs", "install:manpages", "install:translations:manpages", "install:translations:applications" ]
-targets += [ "install:translations:guides", "install:shared" ] if c?(:USE_WXWIDGETS)
+targets += [ "install:translations:guides" ] if c?(:USE_WXWIDGETS)
+targets += [ "install:shared" ]              if c?(:USE_WXWIDGETS) || c?(:USE_QT)
 task :install => targets
 
 namespace :install do
@@ -579,8 +580,10 @@ namespace :install do
 
   task :shared do
     install_dir :desktopdir, :mimepackagesdir
-    install_data :desktopdir, FileList[ "#{$top_srcdir}/share/desktop/*.desktop" ]
     install_data :mimepackagesdir, FileList[ "#{$top_srcdir}/share/mime/*.xml" ]
+    install_data :desktopdir, "#{$top_srcdir}/share/desktop/mkvinfo.desktop"
+    install_data :desktopdir, "#{$top_srcdir}/share/desktop/mkvmergeGUI.desktop"    if c?(:USE_WXWIDGETS)
+    install_data :desktopdir, "#{$top_srcdir}/share/desktop/mkvtoolnix-gui.desktop" if c?(:USE_QT)
 
     wanted_apps     = %w{mkvmerge mkvtoolnix-gui mkvinfo mkvextract mkvpropedit}.collect { |e| "#{e}.png" }.to_hash_by
     wanted_dirs     = %w{16x16 24x24 32x32 48x48 64x64 96x96 128x128 256x256}.to_hash_by
